src/tess_catalog.jl

```diff
@@ -0,0 +1,389 @@
+- `sim_param::SimParam`: Simulation parameter object; requires the following simulation parameters to be set:
+  - calc_target_obs_single_obs: Function name for single observer simulated observations
+
+# Returns:
+TESS observations catalog object containing all properties observed from the TESS targets and associated planetary systems that were detected during the simulation.
+"""
+function observe_tess_targets_single_obs(input::TESSPhysicalCatalog, sim_param::SimParam )
+  calc_target_obs = get_function(sim_param,"calc_target_obs_single_obs")
+  return observe_tess_targets(calc_target_obs, input, sim_param)
+end
+
+"""
+    observe_tess_targets(calc_target_obs, input, sim_param)
+
+Wrapper function to create catalog of simulated observations of TESS targets.
+
+# Arguments:
+- `calc_target_obs::Function`: Function to use in simulating observations of TESS targets (sky averaging vs. single observer schemes)
+- `input::TESSPhysicalCatalog`: Catalog object of simulated TESS targets and associated planetary systems to be observed
+- `sim_param::SimParam`: Simulation parameter object
+
+# Returns:
+TESS observations catalog object containing all properties observed from the TESS targets and associated planetary systems that were detected during the simulation.
+"""
+function observe_tess_targets(calc_target_obs::Function, input::TESSPhysicalCatalog, sim_param::SimParam )
+  #calc_target_obs = get_function(sim_param,"calc_target_obs_sky_ave")
+  #calc_target_obs = get_function(sim_param,"calc_target_obs_single_obs")
+  output = TESSObsCatalog()
+  if haskey(sim_param,"mem_tess_target_obs")
+     output.target = get(sim_param,"mem_tess_target_obs",Array{TESSTargetObs}(0) )
+  end
+  num_targets_sim_pass_one = get_int(sim_param,"num_targets_sim_pass_one")
+  if length(output.target) < num_targets_sim_pass_one
+     output.target = Array{TESSTargetObs}(undef,num_targets_sim_pass_one)
+  end
+  map!(x::TESSTarget->calc_target_obs(x,sim_param)::TESSTargetObs, output.target, input.target)
+  resize!(output.target,length(input.target))
+  return output
+end
+
+# Test if this planetary system has at least one planet that transits (assuming a single observer)
+function select_targets_one_obs(ps::PlanetarySystemAbstract)
+ for pl in 1:length(ps.orbit)
+   ecc::Float64 = ps.orbit[pl].ecc
+   incl::Float64 = ps.orbit[pl].incl
+   a::Float64 = semimajor_axis(ps,pl)
+   Rstar::Float64 = rsol_in_au*ps.star.radius
+   if (Rstar > (a*(1-ecc)*(1+ecc))/(1+ecc*sin(ps.orbit[pl].omega))*cos(incl))
+     return true
+   end
+ end
+ return false
+end
+#=
+function select_targets_one_obs(ps::PlanetarySystemAbstract)
+  for pl in 1:length(ps.orbit)
+    if does_planet_transit(ps,pl)
+       return true
+    end
+  end
+  return false
+end
+=#
+
+# Remove undetected planets from physical catalog
+# TODO: OPT: Maybe create array of bools for which planets to keep, rather than splicing out non-detections?
```

Comment:
FYI, We found that for some parameters, one of our models would generate lots of multiple planet systems where only one or a few transited.  The resizing was meant to help with memory requirements.